### PR TITLE
Murmur3

### PR DESCRIPTION
**Pull request by neilstevenson**
Adds the blog post text to README.md

### CODE REVIEW

File: serialization/murmur3/README.md [hunk 1]
```diff
@@ -12,4 +10,184 @@ Run with:
...
+
+In a Hazelcast grid, there are usually several nodes doing data storage.
+
+In this case, the <code>IMap</code> would be spread across these processes. Each process hosts a share of the map's contents.
+
+This gives the obvious benefit of scalability. With three nodes, each node hosts a third of the map. So the map's capacity is triple the capacity of a single process. If we add a fourth process, the map's capacity is now four times that of a single process. We have scaled up.
+
+However, there is also the potential for confusion if you don't understand some of the necessary implementation details.
+
+For example, with 3 processes, each has ⅓ of the map to look after but it might not mean ⅓ of the data.
```
Comment (mmedenjak) [inline serialization/murmur3/README.md:49]: Maybe `with 3 processes, each has ⅓ of the key range to look after`?

File: serialization/murmur3/README.md [hunk 2]
```diff
@@ -12,4 +10,184 @@ Run with:
...
+
+One last point to look at here is keys 15 and 16.
+<pre>Key '15', hashcode 1398412787, is in partition 213, hosted currently by node node1
+Key '16', hashcode 1099026892, is in partition 213, hosted currently by node node1
+</pre>
+They have different hash codes. When the hash code has modulus 271 applied, keys will end up sharing partitions as we have more keys than partitions. Sometimes this will be consecutive keys.
+
+Since keys 15 and 16 are in the same partition, they will always be on the same node, regardless of the cluster size.
+<h3>Checkpoint</h3>
+As` a bonus, there is one last problem.
```
Comment (mmedenjak) [inline serialization/murmur3/README.md:170]: Unnecessary backtick

File: serialization/murmur3/README.md [hunk 3]
```diff
@@ -12,4 +10,184 @@ Run with:
...
+So now we have an explanation.
+
+Code below will show that the keys 1, 2, 5, 9 and 50 are assigned to
+partitions 31, 5, 169, 42 and 105 respectively.
+
+In our 3-nodes cluster, the 271 partitions mean the nodes get 90 or 91 partitions each, the best we can spread them out. Recall that when we do this, we don't know what data will be inserted.
+
+At this point, it's not unreasonable that the node on the left of the diagram above owns partitions 5, 31 and 42, along with nearly ninety other partitions.
+
+In fact, if we only inserted records with keys 1, 2 and 9, we would find all our data on one node.
```
Comment (mmedenjak) [inline serialization/murmur3/README.md:124]: You may mention if you want that this is an example of a "skewed" distribution.

File: serialization/murmur3/README.md [hunk 4]
```diff
@@ -12,4 +10,184 @@ Run with:
...
+
+Even if we had 25 Hazelcast servers and it happened that they had 1 of the 25 keys each, there would still be an imbalance as the data records are not the same size.
+
+Normally you would expect variable-sized records to vary within a range rather than continually get larger, but this is a feature of the application. Hazelcast stores what it is given.
+<h2>Summary</h2>
+The use of prime numbers and hashing algorithms make it reasonably likely that a Hazelcast cluster will remain balanced for large data volumes.
+
+It is however not a guarantee.
+
+When having strange patterns to key allocation, it is possible that data allocation might not be uniform. Some partitions may get more records than others.
```
Comment (mmedenjak) [inline serialization/murmur3/README.md:189]: Again, you may mention skewed distribution.

File: serialization/murmur3/README.md [hunk 5]
```diff
@@ -12,4 +10,184 @@ Run with:
...
 java -jar target/murmur3-0.1-SNAPSHOT.jar 
-```
\ No newline at end of file
+```
+
+# Where Are My Keys ?
+
+It’s useful to understand how, why and where Hazelcast stores your data in the grid.
+
+What happens is what you really need, but it won’t necessarily be what you first think you want.
```
[the remark this answers is not in the document]
Comment (neilstevenson) [reply serialization/murmur3/README.md:19]: It's a cut&paste from https://hazelcast.com/blog/where-are-my-keys/,  I'd not want to change it so the two match

File: serialization/murmur3/README.md (hunk 1 above)
Comment (neilstevenson) [reply serialization/murmur3/README.md:49]: As above, a cut&paste from https://hazelcast.com/blog/where-are-my-keys/, don't want to amend to keep the two matching

File: serialization/murmur3/README.md (hunk 3 above)
Comment (neilstevenson) [reply serialization/murmur3/README.md:124]: As above, a cut&paste from https://hazelcast.com/blog/where-are-my-keys/, don't want to amend to keep the two matching

File: serialization/murmur3/README.md (hunk 4 above)
Comment (neilstevenson) [reply serialization/murmur3/README.md:189]: As above, a cut&paste from https://hazelcast.com/blog/where-are-my-keys/, don't want to amend to keep the two matching

File: serialization/murmur3/README.md [hunk 6]
```diff
@@ -12,4 +10,184 @@ Run with:
...
+
+Our data record has a primary key. We wish to retrieve using the primary key. This is what a map does best.
+<h2>IMap</h2>
+<a href="https://docs.hazelcast.org/docs/4.0.1/javadoc/com/hazelcast/map/IMap.html">IMap</a> is Hazelcast’s implementation of the classic map data structure.
+
+In a Hazelcast grid, there are usually several nodes doing data storage.
+
+In this case, the <code>IMap</code> would be spread across these processes. Each process hosts a share of the map's contents.
+
+This gives the obvious benefit of scalability. With three nodes, each node hosts a third of the map. So the map's capacity is triple the capacity of a single process. If we add a fourth process, the map's capacity is now four times that of a single process. We have scaled up.
```
[the remark this answers is not in the document]
Comment (neilstevenson) [reply serialization/murmur3/README.md:45]: As above, a cut&paste from https://hazelcast.com/blog/where-are-my-keys/, don't want to amend to keep the two matching

File: serialization/murmur3/README.md [hunk 7]
```diff
@@ -12,4 +10,184 @@ Run with:
...
\ No newline at end of file
+```
+
+# Where Are My Keys ?
+
+It’s useful to understand how, why and where Hazelcast stores your data in the grid.
+
+What happens is what you really need, but it won’t necessarily be what you first think you want.
+
+We’ll explore this in this blog by putting some number keys in a map, a common starting point and source of confusion.
```
[the remark this answers is not in the document]
Comment (neilstevenson) [reply serialization/murmur3/README.md:21]: As above, a cut&paste from https://hazelcast.com/blog/where-are-my-keys/, don't want to amend to keep the two matching

File: serialization/murmur3/README.md (hunk 2 above)
Comment (neilstevenson) [reply serialization/murmur3/README.md:170]: Will push a fix

Comment (neilstevenson) [reply serialization/murmur3/README.md]: Fix pushed

File: serialization/murmur3/README.md (hunk 5 above)
[the remark this answers is not in the document]
Comment (mmedenjak) [reply serialization/murmur3/README.md:19]: You can change the blog if we agree that the text should be changed. What's the point of a review if I can't request changes?

**Pull request merged**